packages/ext-taquito/src/readProviderWrapper.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  BlockResponse,
+  EntrypointsResponse,
+  MichelsonV1Expression,
+  SaplingDiffResponse, ScriptedContracts,
+} from '@taquito/rpc';
+import {
+  BigMapQuery,
+  BlockIdentifier,
+  SaplingStateQuery,
+  TzReadProvider
+} from '@taquito/taquito';
+import {BigNumber} from 'bignumber.js';
+
+import {TzktExtension} from './index';
+
+
+export class ReadProviderWrapper implements TzReadProvider {
+  constructor(private readProvider: TzReadProvider, private tzktExtension: TzktExtension) {}
```

Comment:
Composition or inheritance?
Either `TzktReadProvider : TzReadProvider` or `constructor(TzReadProvider)` not both

TzktReadProvider <-> TzktExtension circular dependency, read provider must not know about the extension